# Created at 2024-08-05T13:07:52.087
org.testng.TestNGException: 
Cannot instantiate class rahulshettyacademy.TestComponents.Listeners
	at org.testng.internal.objects.InstanceCreator.newInstance(InstanceCreator.java:43)
	at org.testng.internal.objects.InstanceCreator.newInstance(InstanceCreator.java:61)
	at org.testng.ITestObjectFactory.newInstance(ITestObjectFactory.java:10)
	at org.testng.internal.objects.SimpleObjectDispenser.dispense(SimpleObjectDispenser.java:61)
	at org.testng.internal.objects.GuiceBasedObjectDispenser.dispenseObject(GuiceBasedObjectDispenser.java:41)
	at org.testng.internal.objects.GuiceBasedObjectDispenser.dispense(GuiceBasedObjectDispenser.java:31)
	at org.testng.internal.DefaultListenerFactory.createListener(DefaultListenerFactory.java:29)
	at org.testng.TestRunner.initListeners(TestRunner.java:389)
	at org.testng.TestRunner.init(TestRunner.java:349)
	at org.testng.TestRunner.init(TestRunner.java:302)
	at org.testng.TestRunner.<init>(TestRunner.java:176)
	at org.testng.SuiteRunner$DefaultTestRunnerFactory.newTestRunner(SuiteRunner.java:636)
	at org.testng.SuiteRunner.<init>(SuiteRunner.java:197)
	at org.testng.TestNG.createSuiteRunner(TestNG.java:1365)
	at org.testng.TestNG.createSuiteRunners(TestNG.java:1339)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1199)
	at org.testng.TestNG.runSuites(TestNG.java:1112)
	at org.testng.TestNG.run(TestNG.java:1079)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:72)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
	at org.testng.internal.objects.InstanceCreator.newInstance(InstanceCreator.java:40)
	... 24 more
Caused by: java.lang.NullPointerException
	at java.base/java.io.File.<init>(File.java:278)
	at com.aventstack.extentreports.reporter.ExtentSparkReporter.<init>(ExtentSparkReporter.java:51)
	at rahulshettyacademy.resources.ExtentReporterNG.getReportObject(ExtentReporterNG.java:14)
	at rahulshettyacademy.TestComponents.Listeners.<init>(Listeners.java:20)
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
	... 27 more


